### PR TITLE
Add ability to manipulate group users

### PR DESCRIPTION
**Pull request by mdellweg**

### CODE REVIEW

File: pulpcore/cli/core/group.py [hunk 1]
```diff
@@ -38,3 +40,38 @@ def create(pulp_ctx: PulpContext, entity_ctx: PulpEntityContext, name: str) -> N
...
+
+
+user.add_command(list_entities)
+
+
+@user.command()
+@click.option("--username", required=True)
+@pass_entity_context
+def add(entity_ctx: PulpGroupUserContext, username: str) -> None:
+    # This href_key seems to be a bug in the apidocs: it should be "auth_group_href"
```
Comment (daviddavis) [inline pulpcore/cli/core/group.py:64]: Was this fixed as part of https://github.com/pulp/pulpcore/pull/1066 ?

Comment (mdellweg) [reply pulpcore/cli/core/group.py:64]: Yes, do we need the version specific workaround, or do we just wait for this to merge?

**Comment by mdellweg** [comment]
Let's postpone this until we can create users and write proper tests for it.

**Comment by mdellweg** [comment]
Actually we have two users to play around with.

File: pulpcore/cli/core/context.py [hunk 2]
```diff
@@ -111,6 +112,21 @@ def find(self, **kwargs: Any) -> Any:
         return search_result[0]
 
 
+class PulpGroupUserContext(PulpEntityContext):
+    ENTITY = "group user"
+    HREF = "auth_auth_groups_user_href"
```
Comment (mdellweg) [inline pulpcore/cli/core/context.py:117]: Reminder to self:
This needs to be changed into a workaround for pulpcore < 3.10

Comment (mdellweg) [reply pulpcore/cli/core/context.py]: done

File: pulpcore/cli/core/context.py [hunk 3]
```diff
@@ -136,6 +136,52 @@ def find(self, **kwargs: Any) -> Any:
...
+
+    def list(self, limit: int, offset: int, parameters: Dict[str, Any]) -> List[Any]:
+        if not self.pulp_ctx.has_plugin("pulpcore", min_version="3.10.dev0"):
+            # Workaround for improperly rendered nested resource paths and weird HREF names
+            # https://github.com/pulp/pulpcore/pull/1066
+            parameters[PulpGroupContext.HREF] = self.group_ctx.pulp_href
+        return super().list(limit=limit, offset=offset, parameters=parameters)
+
+    def create(self, body: EntityDefinition, parameters: Optional[Dict[str, Any]] = None) -> Any:
+        if not self.pulp_ctx.has_plugin("pulpcore", min_version="3.10.dev0"):
```
Comment (daviddavis) [inline pulpcore/cli/core/context.py:160]: Should this be `3.10.0.dev0`?

**Review by daviddavis — APPROVED** [review]
LGTM. one question.

File: pulpcore/cli/core/context.py (hunk 3 above)
Comment (mdellweg) [reply pulpcore/cli/core/context.py:160]: I'd say, it's a matter of taste.

```python
In [1]: from packaging.version import parse

In [2]: parse("3.9.99") < parse("3.10.dev0") == parse("3.10.0.dev0") < parse("3.10")
Out[2]: True
```

**Pull request merged**